maven-embedder/src/main/java/org/apache/maven/cli/MavenCli.java

```diff
@@ -374,25 +376,28 @@ void cli(CliRequest cliRequest) throws Exception {
         cliManager = new CLIManager();
 
         CommandLine mavenConfig = null;
-        try {
-            File configFile = new File(cliRequest.multiModuleProjectDirectory, MVN_MAVEN_CONFIG);
-
-            if (configFile.isFile()) {
-                try (Stream<String> lines = Files.lines(configFile.toPath(), Charset.defaultCharset())) {
-                    String[] args = lines.filter(arg -> !arg.isEmpty() && !arg.startsWith("#"))
-                            .toArray(String[]::new);
-                    mavenConfig = cliManager.parse(args);
-                    List<?> unrecognized = mavenConfig.getArgList();
-                    if (!unrecognized.isEmpty()) {
-                        // This file can only contain options, not args (goals or phases)
-                        throw new ParseException("Unrecognized maven.config file entries: " + unrecognized);
+        if (cliRequest.rootDirectory != null) {
+            try {
+                File configFile =
```

Comment:
Better switch to nio file api ?

Reply:
Least changes is what I wanted in 3.9, but I can do that (but seems unrelated)

Reply:
Yeah, I only saw it was targeted to 3.9.x later.